repose-aggregator/components/filters/herp/src/test/scala/org.openrepose.filters.herp/HerpFilterTest.scala

```diff
@@ -0,0 +1,335 @@
+import org.apache.logging.log4j.test.appender.ListAppender
+import org.openrepose.filters.herp.config.{Template, HerpConfig}
+import org.scalatest._
+import org.springframework.mock.web.{MockFilterChain, MockHttpServletRequest, MockHttpServletResponse}
+
+import scala.collection.JavaConverters._
+
+class HerpFilterTest extends FunSpec with BeforeAndAfterAll with BeforeAndAfter with Matchers {
+
+  var herpFilter: HerpFilter = _
+  var herpConfig: HerpConfig = _
+  var servletRequest: MockHttpServletRequest = _
+  var servletResponse: MockHttpServletResponse = _
+  var filterChain: MockFilterChain = _
+  var listAppender: ListAppender = _
+
+  override def beforeAll() {
+    System.setProperty("javax.xml.parsers.DocumentBuilderFactory",
+      "com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl");
+  }
+
+  //todo: replace this and the log4j2 config with programmatic modification to the root logger with a mock appender?
+  before {
+    val ctx = LogManager.getContext(false).asInstanceOf[LoggerContext]
+    listAppender = ctx.getConfiguration.getAppender("highly-efficient-record-processor-ListAppender").asInstanceOf[ListAppender].clear
+
+    herpFilter = new HerpFilter
+    herpConfig = new HerpConfig
+    servletRequest = new MockHttpServletRequest
+    servletResponse = new MockHttpServletResponse
+    filterChain = new MockFilterChain
+    herpConfig.setLoggerName("highly-efficient-record-processor-Logger")
+    val templateText =
+      """
+        |{
+        | "GUID" : "{{guid}}",
+        | "ServiceCode" : "{{serviceCode}}",
+        | "Region" : "{{region}}",
+        | "DataCenter" : "{{dataCenter}}",
+        | "Timestamp" : "{{timestamp}}",
+        | "Request" : {
+        |   "Method" : "{{requestMethod}}",
+        |   "URL" : "{{requestURL}}",
+        |   "Parameters" : {
+        |                     {{#parameters}}
+        |                     "{{key}}" : [
+        |                       {{#value}}
+        |                       "{{value}}",
+        |                       {{/value}}
+        |                     ],
+        |                     {{/parameters}}
+        |                  },
+        |   "UserName" : "{{userName}}",
+        |   "ImpersonatorName" : "{{impersonatorName}}",
+        |   "TenantID" : "{{tenantID}}",
+        |   "Roles" : [
+        |               {{#roles}}
+        |               "{{name}}",
+        |               {{/roles}}
+        |             ],
+        |   "UserAgent" : "{{userAgent}}"
+        | },
+        | "Response" : {
+        |   "Code" : {{responseCode}},
+        |   "Message" : "{{responseMessage}}"
+        | }
+        |}
+      """.stripMargin
+    val template = new Template
+    template.setValue(templateText)
+    template.setCrush(true)
+    herpConfig.setTemplate(template)
+  }
+
+  describe("the doFilter method") {
+    it("should be empty if field data is not present") {
+      // when:
```

Comment:
Hah, I was looking for a ScalaTest trait that supported parameterized testing is a nice way, but what I found was kind of ugly. What would these tests look like using the FeatureSpec trait?